packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart

```diff
@@ -948,23 +948,46 @@ class ObjectGroup extends InspectorObjectGroupBase {
   @override
   bool canSetSelectionInspector = true;
 
-  Future<RemoteDiagnosticsNode?> getRoot(FlutterTreeType type) {
+  Future<RemoteDiagnosticsNode?> getRoot(
+    FlutterTreeType type, {
+    required bool isSummaryTree,
+  }) {
     // There is no excuse to call this method on a disposed group.
     assert(!disposed);
     switch (type) {
       case FlutterTreeType.widget:
-        return getRootWidget();
+        return getRootWidgetTree(isSummaryTree: isSummaryTree);
     }
   }
 
-  Future<RemoteDiagnosticsNode?> getRootWidget() {
-    return parseDiagnosticsNodeDaemon(
-      invokeServiceMethodDaemonParams(
-        WidgetInspectorServiceExtensions
-            .getRootWidgetSummaryTreeWithPreviews.name,
-        {'groupName': groupName},
-      ),
-    );
+  Future<RemoteDiagnosticsNode?> getRootWidgetTree({
+    required bool isSummaryTree,
+  }) {
+    const newApi = 'getRootWidgetTree';
+    final supportsNewApi =
+        WidgetInspectorServiceExtensions.values.asNameMap().containsKey(newApi);
+    if (supportsNewApi) {
```

Comment:
Since DevTools is tied to the Flutter SDK, we don't need to worry about backwards compatibility - we can just migrate to the new API. The CI will pass once the flutter candidate check picks up a version of Flutter that includes your new service extension change in flutter/flutter.